scripts/core/editor3/helpers/tansa.js

```diff
@@ -0,0 +1,201 @@
+ * @returns {EditorState}
+ */
+export function setTansaHtml(editorState, html) {
+    let content = editorState.getCurrentContent();
+    const blockMap = content.getBlockMap();
+    const htmlElement = document.createElement('div');
+
+    htmlElement.innerHTML = html;
+
+    blockMap.forEach((block) => {
+        const key = block.getKey();
+
+        if (block.getType() === 'atomic') {
+            const newDescription = getTextFromTag(htmlElement, 'description', key);
+            const newAlt = getTextFromTag(htmlElement, 'alt', key);
+            const newHeadline = getTextFromTag(htmlElement, 'headline', key);
+
+            content = updateMedia(content, block, newDescription, newAlt, newHeadline);
+        } else {
+            const newText = getTextFromTag(htmlElement, 'text', key);
+
+            content = updateText(editorState, content, block, newText);
+        }
+    });
+
+    return EditorState.push(editorState, content, 'insert-characters');
+}
+
+/**
+ * Find the tag and extract the text
+ *
+ * @param {HtmlDom} htmlElement
+ * @param {String} field
+ * @param {String} key
+ * @returns {String}
+ */
+function getTextFromTag(htmlElement, field, key) {
+    const tagElement = htmlElement.querySelector('#' + getHtmlId(field, key));
+
+    return tagElement != null ? tagElement.innerHTML : null;
+}
+
+/**
+ * Generate an id for html tag
+ *
+ * @param {String} field
+ * @param {String} key
+ * @returns {String}
+ */
+function getHtmlId(field, key) {
+    return field + '-' + key;
+}
+
+/**
+ * Generate the tag
+ *
+ * @param {String} field
+ * @param {String} key
+ * @param {String} text
+ * @returns {String}
+ */
+function getBlockHtml(field, key, text) {
+    return `<p id="${getHtmlId(field, key)}">${text}</p>\n`;
+}
+
+/**
+ * Update the description of the media block
+ *
+ * @param {ContentState} contentState
+ * @param {Block} block
+ * @param {String} text
+ * @returns {ContentState}
+ */
+function updateMedia(content, block, newDescription, newAlt, newHeadline) {
+    const entityKey = block.getEntityAt(0);
+    const entity = content.getEntity(entityKey);
+    const data = (entity != null && entity.getData() != null) ? entity.getData() : null;
+    let newContent = content;
+
+    if (data != null) {
+        if (newDescription) {
+            data.media.description_text = newDescription;
+        }
+
+        if (newAlt) {
+            data.media.alt_text = newAlt;
+        }
+
+        if (newHeadline) {
+            data.media.headline = newHeadline;
+        }
+
+        newContent = content.mergeEntityData(entityKey, data);
+    }
+
+    return newContent;
+}
+
+/**
+ * Update the text in the block
+ *
+ * @param {EditorState} editorState
+ * @param {ContentState} contentState
+ * @param {Block} block
+ * @param {String} text
+ * @returns {ContentState}
+ */
+function updateText(editorState, content, block, newText) {
+    const text = block.getText();
+    const diffMatchPatch = new DiffMatchPatch();
```

Comment:
I'm not sure how DiffMatchPatch works but shouldn't we create only one object outside of the function instead of one object per function call? It seems more efficient